backend/audit/views/views.py

```diff
@@ -61,6 +61,34 @@ def _friendly_status(status):
     return dict(SingleAuditChecklist.STATUS_CHOICES)[status]
 
 
+def verify_status(status):
+    """
+    Decorator to be applied to view request methods (i.e. get, post) to verify
+    that the submission is in the correct state before allowing the user to
+    proceed. An incorrect status usually happens via direct URL access. If the
+    given status does not match the submission's, it will redirect them back to
+    the submission progress page.
+    """
+
+    def decorator_verify_status(req_method):
+        def verify(view, request, *args, **kwargs):
+            report_id = kwargs["report_id"]
+            sac = SingleAuditChecklist.objects.get(report_id=report_id)
```

Comment:
what would happen if this look-up fails to find a Single Audit Checklist with that report id?

Reply:
Good catch, checking for that now